packages/plugin-json/src/lib/validation.ts

```diff
@@ -82,6 +83,62 @@ const isDeliveryKeyField = (field: FieldDefinitionNode) =>
 const isNullableStringField = (field: FieldDefinitionNode) =>
   field.type.kind === "NamedType" && field.type.name.value === "String";
 
+export const getAmplienceExtensionNotNullableObjectReport = (
+  types: ObjectTypeDefinitionNode[],
+): string =>
+  getFieldsReport(
+    types.filter(
+      (type) =>
+        type.fields?.some(
+          (field) =>
+            isAmplienceExtensionField(field) &&
+            !isNullableObjectField(field.type),
+        ),
+    ),
+    (field) =>
+      isAmplienceExtensionField(field) && !isNullableObjectField(field.type),
+  );
+
+// @amplienceExtension referencing an @amplienceContentType produces odd behavior we want to avoid
+export const getAmplienceExtensionReferencesAmplienceContentTypeReport = (
+  types: ObjectTypeDefinitionNode[],
+): string =>
+  getFieldsReport(
+    types.filter(
+      (type) =>
+        type.fields?.some(
+          (field) =>
+            isAmplienceExtensionField(field) &&
+            isNullableObjectField(field.type) &&
+            isAmplienceContentTypeField(field.type, types),
+        ),
+    ),
+    (field) =>
+      isAmplienceExtensionField(field) &&
+      isNullableObjectField(field.type) &&
+      isAmplienceContentTypeField(field.type, types),
+  );
+
+const isAmplienceExtensionField = (field: FieldDefinitionNode) =>
+  hasDirective(field, "amplienceExtension");
+
+const isNullableObjectField = (type: TypeNode): type is NamedTypeNode =>
+  type.kind === "NamedType" &&
+  !["Int", "Float", "String", "Boolean", "ID"].includes(type.name.value);
```

Comment:
I think it's better to check if the name is in all types, then you know for sure it's an actual object type.
Because right now, I think it might break for scalars (e.g. Date).

Reply:
Good spot. Updated, and added a test case to catch the scalar example